### PR TITLE
Omnibox and QueryBox now uses disableQuerySyntax instead of putting <…

### PR DESCRIPTION
**Pull request by fgagnoncoveo**
…@- -@> tags

https://coveord.atlassian.net/browse/JSUI-804

### CODE REVIEW

**Comment by fgagnoncoveo** [comment]
@olamothe

File: src/rest/Query.ts [hunk 1]
```diff
@@ -164,6 +164,12 @@ export interface IQuery {
   groupBy?: IGroupByRequest[];
   debug?: boolean;
   timezone?: string;
+  /**
+   * Wheter to disable the special query syntax such as field references for the basic query expression (parameter q).
```
Comment (mcarpentierCoveo) [inline src/rest/Query.ts]: Whether

File: src/ui/Base/QueryBuilder.ts [hunk 2]
```diff
@@ -95,6 +95,12 @@ export class QueryBuilder {
    */
   public enableQuestionMarks: boolean;
   /**
+   * Wheter to disable the special query syntax such as field references for the basic query expression (parameter q).
```
Comment (mcarpentierCoveo) [inline src/ui/Base/QueryBuilder.ts]: Same

File: src/ui/Omnibox/Omnibox.ts [hunk 3]
```diff
@@ -387,11 +387,8 @@ export class Omnibox extends Component {
     }
 
     if (!_.isEmpty(query)) {
-      if (!this.options.enableQuerySyntax) {
-        data.queryBuilder.expression.add("<@- " + query + " -@>");
-      } else {
-        data.queryBuilder.expression.add(query);
-      }
+      data.queryBuilder.disableQuerySyntax = !this.options.enableQuerySyntax;
```
Comment (mcarpentierCoveo) [inline src/ui/Omnibox/Omnibox.ts]: maybe use the same name for the option: data.queryBuilder.enableQuerySyntax

Comment (fgagnoncoveo) [inline src/ui/Omnibox/Omnibox.ts]: The option's called that way in the search api https://developers.coveo.com/display/public/SearchREST/Query+Parameters#QueryParameters-disableQuerySyntax

File: src/ui/Querybox/Querybox.ts [hunk 4]
```diff
@@ -220,11 +220,8 @@ export class Querybox extends Component {
     }
 
     if (!_.isEmpty(this.lastQuery)) {
-      if (!this.options.enableQuerySyntax) {
-        args.queryBuilder.expression.add("<@- " + this.lastQuery + " -@>");
-      } else {
-        args.queryBuilder.expression.add(this.lastQuery);
-      }
+      args.queryBuilder.disableQuerySyntax = !this.options.enableQuerySyntax;
+      args.queryBuilder.expression.add(this.lastQuery);
```
Comment (olamothe) [inline src/ui/Querybox/Querybox.ts]: Missing the check about whether or not the query is empty (adding an empty query in the query builder will break)

**Comment by olamothe** [comment]
:shipit:

**Pull request merged**